staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/typeconverter.go

```diff
@@ -80,16 +81,17 @@ func NewTypeConverter(models proto.Models, preserveUnknownFields bool) (TypeConv
 }
 
 func (c *typeConverter) ObjectToTyped(obj runtime.Object) (*typed.TypedValue, error) {
-	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
```

Comment:
was the output of ObjectToTyped effectively a deep copy before? with these changes, are there cases where it is no longer a deep copy?

Reply:
Yes, it used to be that we always converted structured->unstructured when creating a `TypedValue`. Now, if the `runtime.Object` is structured we wrap it with a low overhead reflection wrapper instead of doing the deep copy to unstructured. The overhead of the reflection wrapper is about 10% CPU and allocations per traversal. And since we only do two traversals, this is much more efficient than doing the deep copy.

If the `runtime.Object` is unstructured, we process it as we did before.

Reply:
so... are there callers that depended on getting a deep copy before because they mutate the result? same question for https://github.com/kubernetes/kubernetes/pull/87908#discussion_r376202279

Reply:
Sorry, I missed the second part of the question. No callers are depending on the deep copy.

`FromStructured` is only called from `ObjectToTyped` which takes a `runtime.Object` that can either be `FromUnstructured` or `FromStructured`.

`ObjectToTyped` callers (excluding tests) are: 
- `versionConverter.Convert`
- `structuredMergeManager.{Update,Apply}`

Neither require a deep copy. And `FromUnstructured` doesn't perform a deep copy, so if clients were to assume a deep copy, they would be in trouble.